### DISCUSSION TITLE
Question about deleting the database

### DISCUSSION THREAD

**Discussion by bgriffy** [amuste/DnetIndexedDb#35] [Q&A] [answered]
I have a method to delete my database: 

![image](https://user-images.githubusercontent.com/11038998/111633456-e846d880-87cb-11eb-834f-2c12db9d27c4.png)

Whenever I call this, it hangs indefinitely. Has anyone had this problem or have any idea as to what I'm doing wrong? 

Some things I've tried: 

- Extracting the call to DeleteIndexedDb() and calling it from outside of the class. 
- Opening the connection to the DB before trying to delete (in this function)

**Comment by amuste** [comment, marked as answer]
Hi @bgriffy 

When you instantiate the Database you can directly call the DeleteIndexedDb() method. 

You have a complete example here:
https://github.com/amuste/DnetIndexedDb/blob/caabaf9b0805537ff8e3709b742c0badba881068/samples/ServerSide/Pages/Index.razor#L47

**Reply by bgriffy** [reply]
Thanks for the reply!

So I have tried directly calling the DeleteIndexedDb method, I.E. here: 

![image](https://user-images.githubusercontent.com/11038998/111828743-1eff1a80-88c2-11eb-8575-9d71415d8c77.png)

I get the same error. Here is where I'm adding the service: 

![image](https://user-images.githubusercontent.com/11038998/111828927-64bbe300-88c2-11eb-9cc5-bf3f715a96c6.png)

I will try to put together a small sample project to see if I can recreate this. Will let you know what I find.

**Reply by bgriffy** [reply]
@amuste Think I have tracked it down. Looks like my delete request was being blocked because I had multiple open connections to the DB, unlike your example where you only have one open connection. The deleteDb func closes out the current connection, but has no way to check for other open connections besides the one thats attempting to delete the DB.

I've made a fork with some proposed changes: 
- Added event listener to listen for when deletion is blocked. 
- Added logic to close connections when OnVersionChange event is fired. 

https://github.com/bgriffy/DnetIndexedDb/commit/b46c8b423de3141be8b7a4e43a79ff8ea61f4365

Let me know if you are good with this so I can make a PR. Thanks!

**Reply by bgriffy** [reply]
@amuste Following up. Are you good with me submitting the PR? 

bgriffy@b46c8b4

**Reply by amuste** [reply]
@bgriffy  Yes. Thanks for the contribution. I will include the change in the next release.

**Reply by bgriffy** [reply]
Sweet, thanks @amuste!  Just submitted PR https://github.com/amuste/DnetIndexedDb/pull/37.
